docs/source/python/extending_types.rst

```diff
@@ -226,6 +226,68 @@ data type from above would look like::
 
 Also the storage type does not need to be fixed but can be parametrized.
 
+Custom extension array class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, all arrays with an extension type are constructed or deserialized into
+a built-in :class:`ExtensionArray` object. Nevertheless, one could want to sub-class
+:class:`ExtensionArray` in order to add some custom logic specific to the extension
+type. Arrow allows to do so by adding a special method ``__arrow_ext_class__`` to the
+definition of the extension type.
+
+For instance, if we consider the case of a 3-dimensional vector, stored as a fixed-size
+list, where we wish to be able to extract the data as a Numpy matrix ``(N, 3)`` without
```

Comment:
I would avoid using the term "matrix", as in context of numpy this can also mean something specific (i.e. the deprecated matrix class). But I think you can say "2-D numpy array"

Reply:
Indeed, my bad about the mixup!